class-instant-articles-post.php

```diff
@@ -759,17 +759,29 @@ public function add_ads_from_settings() {
 
 			case 'embed':
 				if ( ! empty( $settings_ads['embed_code'] ) ) {
+					$sectionAds = $settings_ads['embed_code'];
+					$tokenizeAds = explode("<!-- Embed Ad Block -->", $sectionAds);
+					
+					foreach ($tokenizeAds as $advert) {
+						$ad = Ad::create()
+						->enableDefaultForReuse()
+						->withWidth( $width )
+						->withHeight( $height );
+						$document = new DOMDocument();
+						$fragment = $document->createDocumentFragment();
+						$valid_html = $fragment->appendXML( $advert );
```

Comment:
Would that also ignore empty strings? I would add something like:

```php
$valid_html = $valid_html && ( ! empty( trim( $advert ) ) );
```

So one can start the first add with the comment without causing any trouble (for readability).